### PR TITLE
TextEditor: Prevent drop event in a masked input (T1188135)

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/js/ui/text_box/ui.text_editor.mask.js [hunk 1]
```diff
@@ -144,9 +146,32 @@ const TextEditorMask = TextEditorBase.inherit({
...
+        const useMaskBehavior = this._useMaskBehavior();
+
+        if(!useMaskBehavior) {
+            return;
+        }
+
+        const eventName = addNamespace(DROP_EVENT_NAME, this.NAME);
+        const input = this._input();
+
+        const callback = (e) => {
```
Comment (ksercs) [inline packages/devextreme/js/ui/text_box/ui.text_editor.mask.js]: [code style]
This `callback` as a separate variable does not improve clearness. U can get rid of this variable or rename it, for example `dropHandler`

File: packages/devextreme/js/ui/text_box/ui.text_editor.mask.js [hunk 2]
```diff
@@ -144,9 +146,32 @@ const TextEditorMask = TextEditorBase.inherit({
 
     _onMouseWheel: noop,
 
-    _render: function() {
+    _useMaskBehavior() {
+        return Boolean(this.option('mask'));
+    },
+
+    _attachDragEventHandlers() {
```
Comment (ksercs) [inline packages/devextreme/js/ui/text_box/ui.text_editor.mask.js]: [naming]
```suggestion
    _attachDropEventHandler() {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 3]
```diff
@@ -12,6 +12,9 @@ const { test, module } = QUnit;
 
 const CLEAR_BUTTON_AREA_CLASS = 'dx-clear-button-area';
 
+const DROP_EVENT_NAME = 'drop';
+
+
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]:
```suggestion

```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 4]
```diff
@@ -897,8 +900,20 @@ module('Events', setupModule, () => {
 
         assert.strictEqual(this.instance.option('value'), currentDate, 'value is updated correctly');
     });
-});
 
+    QUnit.test('drag events should be prevented in a masked date box', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]:
```suggestion
    QUnit.test('drop event should be prevented in a masked date box (T1188135)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 5]
```diff
@@ -2409,3 +2411,17 @@ QUnit.module('ShadowDOM', {}, function() {
     });
 });
 
+QUnit.module('drag text', moduleConfig, () => {
+    QUnit.test('drag events should be prevented in a masked number box', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
    QUnit.test('drop event should be prevented in a masked number box (T1188135)', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js [hunk 6]
```diff
@@ -2015,40 +2017,15 @@ QUnit.module('paste', moduleConfig, () => {
...
-    QUnit.test('mask should support drag with spaces', function(assert) {
-        const $textEditor = $('#texteditor').dxTextEditor({
-            mask: '(XXXX)',
-            maskRules: {
-                'X': /[xy]/
-            },
-            value: ' x y'
-        });
-
+    QUnit.test('drag events should be prevented in a masked text box', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js]:
```suggestion
    QUnit.test('drop event should be prevented in a masked text box (T1188135)', function(assert) {
```

File: packages/devextreme/js/ui/text_box/ui.text_editor.mask.js (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/text_box/ui.text_editor.mask.js]: I have used the function for debugging. Now I would rather delete a separate variable. Fixed in [f0b0bf5f2a89e12ddc6f0d76ad534fe221a08aef].

File: packages/devextreme/js/ui/text_box/ui.text_editor.mask.js (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/text_box/ui.text_editor.mask.js]: Fixed in [f0b0bf5f2a89e12ddc6f0d76ad534fe221a08aef].

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js (hunk 3 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: Fixed in [f0b0bf5f2a89e12ddc6f0d76ad534fe221a08aef].

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 5 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: Fixed in [f0b0bf5f2a89e12ddc6f0d76ad534fe221a08aef].

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js (hunk 6 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js]: Fixed in [f0b0bf5f2a89e12ddc6f0d76ad534fe221a08aef].

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js (hunk 4 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: Fixed here [2a71e425809a664788b0a5cf3e78b49e2c488dbd].

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by alexanderPolosatov — APPROVED** [review]

**Pull request merged**